modules/transcription-service-ibm-watson-impl/src/main/java/org/opencastproject/transcription/ibmwatson/IBMWatsonTranscriptionService.java

```diff
@@ -872,16 +875,32 @@ public void run() {
 
       try {
         // Find jobs that are in progress and jobs that had transcription complete i.e. got the callback
-        List<TranscriptionJobControl> jobs = database.findByStatus(TranscriptionJobControl.Status.Progress.name(),
+
+        long providerId;
+        TranscriptionProviderControl providerInfo = database.findIdByProvider(PROVIDER);
+        if (providerInfo != null) {
+          providerId = providerInfo.getId();
+        } else {
+          logger.debug("No provider entry for {}", PROVIDER);
+          return;
+        }
+
+        List<TranscriptionJobControl> jobs = database.findByStatus(TranscriptionJobControl.Status.InProgress.name(),
                 TranscriptionJobControl.Status.TranscriptionComplete.name());
 
         for (TranscriptionJobControl j : jobs) {
+
+          // Don't process jobs for other services
+          if (j.getProviderId() != providerId) {
```

Comment:
While i'm sure there should not be *that* many jobs, I think having the query itself do this filtering probably makes more sense.  Perhaps in a follow on PR.

Reply:
Agreed (for a later PR), at which point Watson, Google Speech and Nibity modules can all be updated to match.